app/src/main/java/org/eyeseetea/malariacare/data/sync/exporter/ConvertToSDKVisitor.java

```diff
@@ -307,26 +318,34 @@ public void saveSurveyStatus(Map<String, ImportSummary> importSummaryMap) {
             Log.d(TAG, "saveSurveyStatus: Starting saving survey Set Survey status as QUARANTINE"
                     + iSurvey.getId_survey() + " eventuid: " + iSurvey.getEventUid());
             iSurvey.save();
-            if(importSummaryMap==null) {
+            if (importSummaryMap == null) {
                 continue;
             }
             EventExtended iEvent = new EventExtended(events.get(iSurvey.getId_survey()));
             ImportSummary importSummary = importSummaryMap.get(iEvent.getEvent().getUId());
-            FailedItemFlow failedItem = EventExtended.hasConflict(iEvent.getLocalId());
+            List<Conflict> conflicts = importSummary.getConflicts();
             //Sets the survey status as quarantine to prevent wrong importSummaries (F.E. in
             // network failures).
             //This survey will be checked again in the future push to prevent the duplicates
             // in the server.
             //If the importSummary has a failedItem the survey was saved in the server but
             // never resend, the survey is saved as survey in conflict.
-            if (failedItem != null) {
-                Log.d(TAG, "saveSurveyStatus: Faileditem not null " + iSurvey.getId_survey());
-                List<String> failedUids = getFailedUidQuestion(failedItem.getErrorMessage());
-                for (String uid : failedUids) {
-                    Log.d(TAG, "saveSurveyStatus: PUSH process...Conflict in " + uid
-                            + " dataelement pushing survey: "
-                            + iSurvey.getId_survey());
-                    iSurvey.setStatus(Constants.SURVEY_CONFLICT);
+            if (conflicts != null && conflicts.size() > 0) {
+                for (Conflict conflict : conflicts) {
+                    Log.d(TAG, "saveSurveyStatus: Faileditem not null " + iSurvey.getId_survey());
+                    if (conflict.getObject() != null) {
+                        Log.d(TAG, "saveSurveyStatus: PUSH process...Conflict in "
+                                + conflict.getObject() +
+                                " with error " + conflict.getValue()
+                                + " dataelement pushing survey: "
+                                + iSurvey.getId_survey());
+                        DashboardActivity.showException(
```

Comment:
@idelcano ConvertToSDKVisitor.java is a object from data Layer, According Clean Architecture data layer should not know presentation layer. Better throw domain exception that catch in pushcontroller and invoke specific error callback then in presentation layer show exception